modules/kafka/kafka_test.go

```diff
@@ -190,6 +190,117 @@ func TestKafka_networkConnectivity(t *testing.T) {
 	}
 }
 
+func TestKafka_listenersValidation(t *testing.T) {
```

Comment:
If you need to test the `trimValidateListeners` function at the unit level, because this file is in the `kafka_test` package, you can create another test file for the validation, with access to the private functions of the package. Then you can just run the tests without creating the container, and just doing real unit-testing on that function. Wdyt?

Reply:
I guess it's unnecessary as it's already pretty fast test and just higher type in pyramid of testing 🤔

Reply:
Yeah, the tests are really fast, I expressed my suggestion about being possible to test that function without a container, just building the struct with the needed values. It's not a blocker, it's just I prefer following the pyramid bottom-up: writing a unit test first if possible, then an integration one, and finally and e2e. But again, not a blocker.